src/lang/uk.rs

```diff
@@ -0,0 +1,1233 @@
+            ADJECTIVE_HARD_FLEXIONS_SINGULAR[Gender::Feminine.index()]
+        }[whole_properties.declenation.index()];
+        
+        let whole_lang = whole_properties.feminine();
+        let numerator_properties = self.agreement_with_num(numerator);
+        let numerator_lang = numerator_properties.feminine();
+        Ok(format!("{} ціл{} {} {}",
+            whole_lang.int_to_cardinal(whole)?,
+            whole_flexion,
+            numerator_lang.int_to_cardinal(numerator)?,
+            numerator_lang.to_ordinal(denominator)?,
+        ))
+    }
+
+    fn ordinal_flexion(&self, num: BigFloat) -> &'static str {
+        let tail = (num % BigFloat::from(100)).to_u64().unwrap();
+        let is_soft = tail%10==3 && tail!=13; //третій - the only soft adjective in numbers
+        let f= match (self.is_plural(), is_soft) {
+            (true, true) => &ADJECTIVE_SOFT_FLEXIONS_PLURAL,
+            (true, false) => &ADJECTIVE_HARD_FLEXIONS_PLURAL,
+            (false, true) => &ADJECTIVE_SOFT_FLEXIONS_SINGULAR[self.gender.index()],
+            (false, false) => &ADJECTIVE_HARD_FLEXIONS_SINGULAR[self.gender.index()],
+        };
+        f[self.declenation.index()]
+    }
+
+    fn ordinal_flexion_short(&self, num: BigFloat) -> &'static str {
+        let f = if self.is_plural() {
+            &ORDINAL_FLEXIONS_PLURAL_SHORT
+        } else{
+            let tail = (num % BigFloat::from(100)).to_u64().unwrap();
+            if tail%10==3 && tail!=13 { //третій - the only soft adjective in numbers        
+                &ORDINAL_SOFT_FLEXIONS_SINGULAR_SHORT[self.gender.index()]
+            } else {
+                &ORDINAL_HARD_FLEXIONS_SINGULAR_SHORT[self.gender.index()]
+            }
+        };
+        f[self.declenation.index()]
+    }
+}
+
+impl Language for Ukrainian {
+    fn to_cardinal(&self, num: BigFloat) -> Result<String, Num2Err> {
+        if num.is_inf_pos() {
+            Ok(String::from(INFINITY[self.declenation.index()]))
+        } else if num.is_inf_neg() {
+            Ok(format!(
+                "мінус {}",
+                INFINITY[self.declenation.index()]
+            ))
+        } else if num.frac().is_zero() {
+            self.int_to_cardinal(num)
+        } else {
+            self.float_to_cardinal(num)
+        }
+    }
+
+    fn to_ordinal(&self, mut num: BigFloat) -> Result<String, Num2Err> {
+        let flexion = self.ordinal_flexion(num);
+
+        // special case zero
+        if num.is_zero() {
+            return Ok(format!("{ORDINAL_ZERO_BASE}{flexion}"));
+        }
+
+        // handling negative values
+        let mut words = vec![];
+        if num.is_negative() {
+            words.push(String::from(MINUS));
+            num = -num;
+        }
+
+        let triplets = self.split_thousands(num);
+        let last_non_empty = triplets.iter().position(|&t|t!=0).unwrap();
+
+        //special case: one unit (thousand, million etc.) in the number
+        if      last_non_empty>0 
```

Comment:
remove the bunch of spaces after `if`.

do you think you can run a format file? this will take care of that for you

Reply:
It looks like those rustfmt::skip with extra spaces after the line somehow broke the formatter. Fixed.
Probably I should file a bug report to rustfmt :)